src/mmw/js/src/draw/views.js

```diff
@@ -1050,6 +1064,185 @@ var WatershedDelineationView = DrawToolBaseView.extend({
+            self.model.reset();
+        });
+    },
+
+    enableStreamSelection: function() {
+        var self = this,
+            map = App.getLeafletMap(),
+            revertLayer = clearAoiLayer();
+
+        // Enable streams layer so user can draw a polygon around a stream
+        var streamLayerCode = 'nhd_streams_hr_v1',
+            collection = App.getLayerTabCollection(),
+            leafletMap = App.getLeafletMap(),
+            layer = collection.findLayerWhere({ code: streamLayerCode }),
+            group = collection.findLayerGroup(layer.get('layerType'));
+
+        if (!layer.get('active')) {
+            coreUtils.toggleLayer(layer, leafletMap, group);
+        }
+
+        utils.drawPolygon(map)
+            .then(validateShape)
+            .then(function(shape) {
+                return self.requestDrainageArea(drawUtils.STREAM, shape);
+            })
+            .then(function(response) {
+                var additionalShapes = {
+                    type: 'FeatureCollection',
+                    id: drawUtils.STREAM,
+                    features: [response.stream_segment],
+                };
+
+                App.map.set('areaOfInterestAdditionals', additionalShapes);
+                addLayer(response.area_of_interest, 'Stream-based Drainage Area');
+                navigateToAnalyze();
+                window.ga('send', 'event', GA_AOI_CATEGORY, 'aoi-create', 'drainage-stream');
+            }).fail(function(message) {
+                revertLayer();
+                displayAlert(message, modalModels.AlertTypes.error);
+                self.model.reset();
+            });
+    },
+
+    // Start polling for a drainage area
+    // `type` should be one of "point" or "stream"
+    // `data` should be a GeoJSON point for "point", a GeoJSON polygon for "stream"
+    requestDrainageArea: function(type, data) {
+        var self = this,
+            deferred = $.Deferred(),
+            taskHelper = {
+                onStart: function() {
+                    self.model.set('polling', true);
+                },
+
+                startFailure: function(response) {
+                    self.model.set({
+                        pollError: true,
+                        polling: false,
+                    });
+                    console.error(response.error);
+                    deferred.reject('Failed to request drainage area');
+                },
+
+                pollSuccess: function(response) {
+                    self.model.set('polling', false);
+
+                    deferred.resolve(response.result);
+                },
+
+                pollFailure: function(response) {
+                    self.model.set({
+                        pollError: true,
+                        polling: false,
+                    });
+                    console.error(response.error);
+                    deferred.reject('Drainage area could not be calculated');
```

Comment:
I noticed that when the polygon is not intersecting with any streams, the app will return the error here, as shown below:
![2022-07-06 16 16 15](https://user-images.githubusercontent.com/60887686/177658623-01c7cca0-5bb4-4c79-a123-11b3527486c0.gif)

I'm thinking would it be helpful if we add something like "the drainage area must intersect with streams" to help first-time users?

Reply:
Thanks for prompting this. Fixed in eb2c372:

![image](https://user-images.githubusercontent.com/1430060/177674931-de9e871f-9f72-411e-b5c4-b570d204bf4f.png)